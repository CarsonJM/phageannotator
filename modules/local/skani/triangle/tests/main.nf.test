nextflow_process {

    name "Test Process SKANI_TRIANGLE"
    script "../main.nf"
    process "SKANI_TRIANGLE"
    config "./nextflow.config"


    test("fasta.gz") {
        when {
            process {
                """
                input[0] = Channel.of(
                    [
                        [ id:'test' ],
                        file("$projectDir/assets/test-datasets/fastas/MEGAHIT-BUFF_01_B.contigs.fa.gz", checkIfExists: true)
                    ]
                )
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("fasta.gz - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = Channel.of(
                    [
                        [ id:'test' ],
                        file("$projectDir/assets/test-datasets/fastas/MEGAHIT-BUFF_01_B.contigs.fa.gz", checkIfExists: true)
                    ]
                )
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
