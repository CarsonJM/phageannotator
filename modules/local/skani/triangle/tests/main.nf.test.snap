{
    "fasta.gz - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,957aeae70193760b192e83610e469118"
                ],
                "triangle": [
                    [
                        {
                            "id": "test"
                        },
                        "test.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,957aeae70193760b192e83610e469118"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-02T10:32:31.871789783"
    },
    "fasta.gz": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.tsv:md5,02f07c89daeaae6b760dc88ba42f2d40"
                    ]
                ],
                "1": [
                    "versions.yml:md5,957aeae70193760b192e83610e469118"
                ],
                "triangle": [
                    [
                        {
                            "id": "test"
                        },
                        "test.tsv:md5,02f07c89daeaae6b760dc88ba42f2d40"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,957aeae70193760b192e83610e469118"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-02T10:32:24.638790526"
    }
}